app/frontend/src/addVacancyStateToDataLayer.js

```diff
@@ -0,0 +1,37 @@
+export const addVacancyStateToDataLayer = {
+  getNewVacancyState() {
+    const newVacancyForm = document.querySelector('.new_job_specification_form');
+    const newVacancyDataset = newVacancyForm && newVacancyForm.dataset;
+    return newVacancyDataset && newVacancyDataset.vacancyState;
+  },
+
+  getCopyVacancyState() {
+    const copyVacancyForm = document.querySelector('.new_copy_vacancy_form');
+    const copyVacancyDataset = copyVacancyForm && copyVacancyForm.dataset;
+    return copyVacancyDataset && copyVacancyDataset.vacancyState;
+  },
+
+  getBodyVacancyDataset() {
+    const bodyVacancyDataset = document.body.dataset;
+    return bodyVacancyDataset
+  },
+
+  getBodyVacancyState() {
+    return this.getBodyVacancyDataset() && this.getBodyVacancyDataset().vacancyState;
+  },
+
```

Comment:
20 lines of code could be 6 as these should all be in the same function as pass in the selector as argument